bindings/objective-c/librpc/librpc.m

```diff
@@ -340,6 +340,10 @@ @implementation RPCListenHandle
 
 - (instancetype)initWithConn:(rpc_connection_t)conn andCookie:(void *)cookie
 {
+    if (!cookie || !conn) {
+        NSLog(@"Could not establish connection. Missing parameters");
```

Comment:
If this function is returning with `nil` and not giving any useful error reporting then maybe it shoud raise an exception or abort?

Reply:
Agree that the error propagation should be addressed, but per discussion with Michael that should be handled in a broader, more consistent manner. I propose leaving it alone for now and opening a ticket.

Reply:
Well, in this case passing wrong parameters to the function is a clear programming error. By programming error I mean an error deliberately not caused by external conditions, but by bug in the code. I think it's fine to convert this if into an assert.

Reply:
Any issue here should be propagated via an NSError so that the user can handle it accordingly. I am happy to do that.

Reply:
In general sense, I don't agree with "leaving this for later". "later" often means "never". Error handling is really critical. We can't just deliberately ignore errors if we want to improve code quality.

Reply:
@michaelfgoldenTwoPG OK, can you jump onto this branch and add the required error handling stuff then? thanks!

Reply:
Looking at the changes Michael is having to do, it is as suspected, very broad and not closely related to the things fixed in this branch. He has a new branch, improve_error_propagation, that he is working on. In the cases for this PR, the nil property error will have already been logged and the code will have returned without attempting to create the RPCListenHandle. The failure that says 'Missing parameters' would only therefore occur if the connection has become null, so perhaps the error message could be made more accurate. The RPCListenHandle is only created for registering event handlers, so it is not leaking out for other calls. Again, I would like to separate the broader work from this PR.

Reply:
@khampton353 @michaelfgoldenTwoPG I agree that every pull request should address exactly one issue. In such case Objective-C code diff should be removed from this PR for two reasons:

* It's an unrelated change
* There's no point in pushing Objective-C changes from this PR only to make new changes on top of them.
* Some changes in this PR are actually a regression in error handling (eg. an assert turned into a log message).

Reply:
Fair enough. These changes will prevent librpc from crashing but not keep cocoapop from crashing until it can handle a null handle. The original urgent ticket was related to cocoapop crashing for whatever reason, but I guess that can just stay open a while longer.